source/utilities.cc

```diff
@@ -547,13 +547,13 @@ namespace aspect
               const double u_theta = spherical_vector[2];
 
               cartesian_vector[0] = std::cos(phi)*std::sin(theta)*u_r
-                                    - std::sin(phi)*u_theta
-                                    - std::cos(phi)*std::cos(theta)*u_phi; // X
+                                    + std::cos(theta)*std::cos(phi)*u_theta
+                                    - std::sin(phi)*u_phi; // X
```

Comment:
Could you make the reordering consistent by also here and below always using `a*u_r + b*u_phi + c*u_theta`, and order the individual terms as `cos(phi)cos(theta)` instead of `cos(theta)cos(phi)`?